taipy/gui/config.py

```diff
@@ -189,7 +189,9 @@ def _handle_argparse(self):
 
         config = self.config
         if args.taipy_port:
-            if not _Config.__RE_PORT_NUMBER.match(args.taipy_port):
+            if str(args.taipy_port).strip() == "auto":
+                config["port"] = "auto"
```

Comment:
I recall we wanted to expose "auto" as a possible value in config files and CLI, but store this as '0' internally, to avoid testing the string over and over...

Reply:
you will still need to test the string regardless since this is too early